### ISSUE TITLE
FR: Add support for Docker/Podman in the install script.

### ISSUE THREAD

**Issue by vega-d** [tailscale/tailscale#5443]
### What are you trying to do?

Currently running the install script (https://tailscale.com/install.sh) on [Fedora Core OS](https://getfedora.org/en/coreos?stream=stable), [Fedora Silverblue](https://silverblue.fedoraproject.org/), Fedora kionite and similar immutable linux distros will just return `sudo: dnf: command not found` error. Following [fedora-specific instructions](https://tailscale.com/download/linux/fedora) (with some adjustments such as replacing `dnf` with `rpm-ostree` and following rpm-ostree specific instructions on adding fedora repos to the system (put .repo file into `/etc/yum.repos.d/`) also doesn't work. 

### How should we solve this?

What does work, however, is docker. Or podman. Both work fine. 

Using [the official docker image](https://hub.docker.com/r/tailscale/tailscale) and instructions posted there work just fine even with podman since by default Fedora Core OS and Silverblue alias `podman=docker`. 

I would love to see the install script detect that there's no DNF package manager, and fall back to docker/podman automatically. 

### What is the impact of not solving this?

Right now there are no obvious links to the docker container [on the main downloads page](https://tailscale.com/download/), nor in [the documentation ](https://tailscale.com/kb/installation/). Unless user will specifically search for the docker container the only way to find out about it's existence is accidentally stumbling upon it while setting up a [docker extension](https://tailscale.com/kb/1184/docker-desktop/#installing-the-tailscale-extension).

Just a couple lines like this: 
```
docker run -d --name=tailscaled -v /var/lib:/var/lib -v /dev/net/tun:/dev/net/tun --network=host --privileged tailscale/tailscale tailscaled

echo "alias tailscale=\"docker exec tailscaled tailscale\"" > ~/.bashrc
```

will basically "install" tailscale just fine. Currently using this setup on my Silverblue and Core OS machines. 

### Anything else?

An alternative solution may be adding a way to discover the docker container, perhaps by doing something like this? (pardon me drawing skills)

![image](https://user-images.githubusercontent.com/35414314/186928041-ff03cdb6-10eb-4085-bac1-9b0510f9871b.png)

**Comment by DentonGentry** [comment]
To set expectations: I doubt very much that we will make the install.sh script start making different changes to the system to run containers if there is no package manager present.

**Comment by vega-d** [comment]
Turns out I was unable to install tailscale rpm because of an upstream [fedora problem which since then has been fixed](https://fedoramagazine.org/manual-action-required-to-update-fedora-silverblue-kinoite-and-iot-version-36/). 

I installed it as a native rpm and there are both ups and downs compared to a docker container: 

- now, MagicDNS actually works. 
- [but, tailscale's repo file breaks graphical app store, `gnome-software`. ](https://github.com/fedora-silverblue/issue-tracker/issues/295)
![image](https://user-images.githubusercontent.com/35414314/189329770-1b84f43b-2c8c-4ebf-8242-51a266ef0807.png)

But I suppose this should be it's own separate issue I am going to file now.
